Content.Server/GameObjects/Components/Atmos/FirelockComponent.cs

```diff
@@ -87,11 +87,13 @@ public override async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
                 }
 
                 if (!await tool.UseTool(eventArgs.User, Owner, holdingPressure || holdingFire ? 1.5f : 0.25f, ToolQuality.Prying)) return false;
-
-                if (State == DoorState.Closed && !IsWeldedShut)
-                    Open();
-                else if (State == DoorState.Open && !IsWeldedShut)
-                    Close();
+                if (!IsWeldedShut) { // Maybe just return here when it's welded?
```

Comment:
mmmm, yeah. Maybe add this condition to the if check above instead of here. That wait you won't get the do_After when the firelock is obviously welded.

Reply:
Not sure exactly what you mean here

Reply:
Instead of doing this if check just after the one before, you can merge both. So it'd check if it's welded shut && then the do after